File: HomeWorks/src/Lesson6/Legs/SonyLeg.java
package Lesson6.Legs;


public class SonyLeg implements ILeg {
    private int price;

    public SonyLeg() {}

    public SonyLeg (int price) {
        this.price = price;
    }

    @Override
    public void step() {
        System.out.println("Шагает нога Sony");
    }

    public void setPrice(int price) {
        this.price = price;
    }

    @Override
    public int getPrice() {
        return price;
    }
}

File: HomeWorkKotlin/src/Robot/Legs/SonyLeg.kt
package Robot.Legs

class SonyLeg(private val priceParam : Int) : ILeg {

    override var price : Int = priceParam
        set(value) {
            if(value > 0) field = value
        }

    override fun step() {
        println("Шагает нога Sony")
    }
}